src/Symfony/Component/Asset/Packages.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset;
+
+/**
+ * Helps manage asset URLs.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Kris Wallsmith <kris@symfony.com>
+ */
+class Packages
+{
+    private $defaultPackage;
+    private $packages = array();
+
+    /**
+     * @param PackageInterface   $defaultPackage The default package
+     * @param PackageInterface[] $packages       Additional packages indexed by name
+     */
+    public function __construct(PackageInterface $defaultPackage = null, array $packages = array())
+    {
+        $this->defaultPackage = $defaultPackage;
+
+        foreach ($packages as $name => $package) {
+            $this->addPackage($name, $package);
+        }
+    }
+
+    /**
+     * Sets the default package.
+     *
+     * @param PackageInterface $defaultPackage The default package
+     */
+    public function setDefaultPackage(PackageInterface $defaultPackage)
+    {
+        $this->defaultPackage = $defaultPackage;
+    }
+
+    /**
+     * Adds a  package.
+     *
+     * @param string           $name    The package name
+     * @param PackageInterface $package The package
+     */
+    public function addPackage($name, PackageInterface $package)
+    {
+        $this->packages[$name] = $package;
+    }
+
+    /**
+     * Returns an asset package.
+     *
+     * @param string $name The name of the package or null for the default package
+     *
+     * @return PackageInterface An asset package
+     *
+     * @throws \InvalidArgumentException If there is no package by that name
```

Comment:
I suggest using a named exception here (extending ``\InvalidArgumentException``) to allow it to be catched specifically